`Not found pair character for [.` when using new highlight blockquote syntax

以下のように[新しくなったハイライトシンタックス](https://github.com/orgs/community/discussions/16925)を利用してtextlintを実行すると、

```md
> [!NOTE]
> some content
```

Not found pair characterが発生しました。

直るとうれしいです 🙏 

```txt
  125:4  error  Not found pair character for [.
                    
You should close this sentence with ].
This pair mark is called angled bracket[]  ja-technical-writing/no-unmatched-pair
```

--- a/package.json
+++ b/package.json
@@ -59,7 +59,7 @@
     ]
   },
   "dependencies": {
-    "sentence-splitter": "^3.0.11",
-    "textlint-rule-helper": "2.0.1"
+    "sentence-splitter": "^4.3.0",
+    "textlint-rule-helper": "^2.3.0"
   }
 }

--- a/src/textlint-rule-no-unmatched-pair.js
+++ b/src/textlint-rule-no-unmatched-pair.js
@@ -1,5 +1,5 @@
 // MIT © 2018 azu
-import { splitAST, Syntax as SentenceSyntax } from "sentence-splitter";
+import { splitAST, SentenceSplitterSyntax } from "sentence-splitter";
 import { PairMaker } from "./parser/PairMaker.js";
 import { SourceCode } from "./parser/SourceCode.js";
 import { IgnoreNodeManager } from "textlint-rule-helper";
@@ -19,7 +19,7 @@ const report = context => {
                 Syntax.BlockQuote,
                 Syntax.Comment
             ]);
-            sentences.children.filter(node => node.type === SentenceSyntax.Sentence).forEach(sentence => {
+            sentences.children.filter(node => node.type === SentenceSplitterSyntax.Sentence).forEach(sentence => {
                 const source = new SourceCode(sentence.raw);
                 const pairMaker = new PairMaker();
                 const sentenceIndex = sentence.range[0];

--- a/test/textlint-rule-no-unmatched-pair-test.js
+++ b/test/textlint-rule-no-unmatched-pair-test.js
@@ -31,7 +31,11 @@ tester.run("textlint-rule-no-unmatched-pair", rule, {
 1) item 1
 2) item 2
 3) item 3
-        `
+        `,
+        // GitHub Flavored Markdown
+        // https://github.com/orgs/community/discussions/16925
+        `> [!NOTE]
+> some content`
     ],
     invalid: [
         {

